src/front/ModuleExpander.hs

```diff
@@ -121,9 +121,13 @@ buildModulePath (NSExplicit ns) =
 findSource :: [FilePath] -> FilePath -> ImportDecl -> IO FilePath
 findSource importDirs sourceDir Import{itarget} = do
   let modulePath = buildModulePath itarget
-      sources = nub $
-                sourceDir </> modulePath :
-                map (</> modulePath) importDirs
+      imports = map (</> modulePath) importDirs
+      sourceModulePath = sourceDir </> modulePath 
+  expandedSourceModulePath <- makeAbsolute $ sourceModulePath
+  let sources = if expandedSourceModulePath `elem` imports then -- in std library directory
```

Comment:
Technically, it checks that the source directory (of the compilation target) is not among the import directories (including the std libs), and if it is it removes it. 
This is a more robust solution, IMO, but I should document the code to reflect this fact.